lib/matplotlib/widgets.py

```diff
@@ -2595,3 +2598,190 @@ def onmove(self, event):
             self.canvas.blit(self.ax.bbox)
         else:
             self.canvas.draw_idle()
+
+
+LABELS_CMAP = ListedColormap(['white', 'red', 'dodgerblue', 'gold',
+                              'greenyellow', 'blueviolet'])
+
+
+class Painter(_SelectorWidget):
+    """
+    Paint regions on an axes.
+
+    For the widget to remain responsive you must keep a reference to it.
+
+    Example usage::
+
+        import numpy as np
+        import matplotlib.pyplot as plt
+        from matplotlib.widgets import Painter
+
+        data = np.random.rand(100, 2)
+        data[:, 1] *= 2
+
+        fig, ax = plt.subplots()
+        pts = ax.scatter(data[:, 0], data[:, 1], s=80)
+
+        def test(x, y):
+            print(x, y)
+
+        p = Painter(ax, test)
+        plt.show()
+    """
+
+    def __init__(self, ax, on_select=None, overlay_props=None,
+                 cursor_props=None, radius=5, cmap=LABELS_CMAP,
+                 useblit=True, button=None, state_modifier_keys=None):
+        """
+        Parameters:
```

Comment:
Please use numpydoc style for new documentation. Basically all you need is to add some underlining:
```
Parameters
----------
```
and then remove the unnecessary *emphasis* on the parameter names.